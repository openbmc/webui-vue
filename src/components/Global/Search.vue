<template>
  <div class="search-global">
    <b-form-group
      :label="$t('global.form.search')"
      :label-for="`searchInput-${_uid}`"
      label-class="invisible"
      class="mb-2"
    >
      <b-input-group size="md" class="align-items-center">
        <b-input-group-prepend>
          <icon-search class="search-icon" />
        </b-input-group-prepend>
        <b-form-input
          :id="`searchInput-${_uid}`"
          ref="searchInput"
          v-model="filter"
          class="search-input"
          type="text"
          :aria-label="$t('global.form.search')"
          :placeholder="placeholder"
          @input="onChangeInput"
        >
        </b-form-input>
        <b-button
          v-if="filter"
          variant="link"
          class="btn-icon-only input-action-btn"
          :title="$t('global.ariaLabel.clearSearch')"
          @click="onClearSearch"
        >
          <icon-close />
          <span class="sr-only">{{ $t('global.ariaLabel.clearSearch') }}</span>
        </b-button>
      </b-input-group>
    </b-form-group>
  </div>
</template>

<script>
import IconSearch from '@carbon/icons-vue/es/search/16';
import IconClose from '@carbon/icons-vue/es/close/20';

export default {
  name: 'Search',
  components: { IconSearch, IconClose },
  props: {
    placeholder: {
      type: String,
      default: function () {
        return this.$t('global.form.search');
      },
    },
  },
  data() {
    return {
      filter: null,
    };
  },
  methods: {
    onChangeInput() {
      this.$emit('change-search', this.filter);
    },
    onClearSearch() {
      this.filter = '';
      this.$emit('clear-search');
      this.$refs.searchInput.focus();
    },
  },
};
</script>

<style lang="scss" scoped>
.search-input {
  padding-left: ($spacer * 2);
}
.search-icon {
  position: absolute;
  left: 10px;
  top: 12px;
  z-index: 4;
  stroke: gray('400');
}
</style>
