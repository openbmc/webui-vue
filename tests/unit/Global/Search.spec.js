import { mount, createLocalVue } from '@vue/test-utils';
import Search from '@/components/Global/Search';
import BootstrapVue from 'bootstrap-vue';

const localVue = createLocalVue();
localVue.use(BootstrapVue);

describe('Search.vue', () => {
  const wrapper = mount(Search, {
    localVue,
    mocks: {
      $t: (key) => key,
    },
  });
  it('should exist', () => {
    expect(wrapper.exists()).toBe(true);
  });
  it('should emit change-search on triggering onChangeInput', () => {
    wrapper.find('input').trigger('input');
    expect(wrapper.emitted('change-search')).toHaveLength(1);
  });
  it('should emit clear-search on triggering onClearSearch', async () => {
    await wrapper.setData({ filter: 'true' });
    wrapper.find('button').trigger('click');
    expect(wrapper.emitted('clear-search')).toHaveLength(1);
  });
  it('should render correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
