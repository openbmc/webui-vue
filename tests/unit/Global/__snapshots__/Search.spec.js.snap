// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search.vue should render correctly 1`] = `
<div
  class="search-global"
>
  <div
    class="form-group mb-2"
    id="__BVID__4"
    role="group"
  >
    <label
      class="d-block invisible"
      for="searchInput-3"
      id="__BVID__4__BV_label_"
    >
      global.form.search
    </label>
    <div>
      <div
        class="input-group align-items-center input-group-md"
        role="group"
      >
        <!---->
        <div
          class="input-group-prepend"
        >
          <svg
            aria-hidden="true"
            class="search-icon"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15,14.3L10.7,10c1.9-2.3,1.6-5.8-0.7-7.7S4.2,0.7,2.3,3S0.7,8.8,3,10.7c2,1.7,5,1.7,7,0l4.3,4.3L15,14.3z M2,6.5	C2,4,4,2,6.5,2S11,4,11,6.5S9,11,6.5,11S2,9,2,6.5z"
            />
          </svg>
        </div>
         
        <input
          aria-label="global.form.search"
          class="search-input form-control"
          id="searchInput-3"
          placeholder="global.form.search"
          type="text"
        />
         
        <!---->
        <!---->
      </div>
      <!---->
      <!---->
      <!---->
    </div>
  </div>
</div>
`;
